app/http.php

```diff
@@ -1019,8 +1042,45 @@ function (string $runtimeId, ?string $payload, string $path, string $method, arr
                     ];
                 }
 
-                $stdout = $responseHeaders['x-open-runtimes-logs'] ?? '';
-                $stderr = $responseHeaders['x-open-runtimes-errors'] ?? '';
+                // Extract logs and errors from file based on fileId in header
+                $fileId = $responseHeaders['x-open-runtimes-log-id'] ?? '';
+                $stdout = '';
+                $stderr = '';
+                if (!empty($fileId)) {
+                    $logs = '';
+                    $errors = '';
+
+                    $logFile = '/tmp/'.$runtimeName .'/logs/' . $fileId . '_logs.log';
+                    $errorFile = '/tmp/'.$runtimeName .'/logs/' . $fileId . '_errors.log';
+
+                    $logDevice = getStorageDevice("/");
+
+                    if ($logDevice->exists($logFile)) {
+                        if ($logDevice->getFileSize($logFile) > 5 * 1024 * 1024) {
+                            $maxToRead = 5 * 1024 * 1024; // read max 5 MB of data
+                            $logs = $logDevice->read($logFile, 0, $maxToRead);
+                            $logs .= "\n... Log file has been truncated to 5 MB.";
+                        } else {
+                            $logs = $logDevice->read($logFile);
+                        }
+                    }
+
+                    if ($logDevice->exists($errorFile)) {
+                        if ($logDevice->getFileSize($errorFile) > 5 * 1024 * 1024) {
+                            $maxToRead = 5 * 1024 * 1024; // read max 5 MB of data
+                            $errors = $logDevice->read($errorFile, 0, $maxToRead);
+                            $errors .= "\n... Error file has been truncated to 5 MB.";
+                        } else {
+                            $errors = $logDevice->read($errorFile);
+                        }
+                    }
+
+                    $stdout = $logs;
+                    $stderr = $errors;
+
+                    $logDevice->delete($logFile);
+                    $logDevice->delete($errorFile);
```

Comment:
We should only delete if file exists. Doing those outside of ifs might cause trouble.